modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/dto/v1_0/converter/AccountDTOConverter.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.headless.commerce.admin.catalog.internal.dto.v1_0.converter;
+
+import com.liferay.account.model.AccountEntry;
+import com.liferay.account.service.AccountEntryLocalService;
+import com.liferay.headless.commerce.admin.catalog.dto.v1_0.Account;
+import com.liferay.portal.kernel.model.User;
+import com.liferay.portal.kernel.security.auth.PrincipalThreadLocal;
+import com.liferay.portal.kernel.service.UserLocalService;
+import com.liferay.portal.vulcan.dto.converter.DTOConverter;
+import com.liferay.portal.vulcan.dto.converter.DTOConverterContext;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Danny Situ
+ */
+@Component(
+	property = "dto.class.name=com.liferay.commerce.account.model.AccountEntry",
+	service = DTOConverter.class
+)
+public class AccountDTOConverter
+	implements DTOConverter<AccountEntry, Account> {
+
+	@Override
+	public String getContentType() {
+		return Account.class.getSimpleName();
+	}
+
+	@Override
+	public Account toDTO(DTOConverterContext dtoConverterContext)
+		throws Exception {
+
+		AccountEntry accountEntry;
+
+		if ((Long)dtoConverterContext.getId() == -1) {
+			User user = dtoConverterContext.getUser();
+
+			if (user == null) {
+				user = _userLocalService.getUserById(
+					PrincipalThreadLocal.getUserId());
+			}
+
+			accountEntry = _accountEntryLocalService.getGuestAccountEntry(
+				user.getCompanyId());
+		}
+		else {
```

Comment:
this doesn't look right to me, I'm not sure we need this, why do we have this? is this the eligibility pattern? Can't we just send the account info along with the payload of the ProductConfigurationListAccount?

Reply:
@luca-pellizzon can you please have a look at this too?